openrave-config.cmake.in

```diff
@@ -32,91 +32,78 @@
+set( OpenRAVE_Boost_VERSION "@Boost_MAJOR_VERSION@.@Boost_MINOR_VERSION@" )
+set( OpenRAVE_CXX_FLAGS "-DOPENRAVE_DLL -DOPENRAVE_CORE_DLL" )
+
+find_package(Boost ${OpenRAVE_Boost_VERSION} REQUIRED COMPONENTS filesystem system thread)
+find_package(LibXml2 REQUIRED)
+
+include(${CMAKE_CURRENT_LIST_DIR}/openrave-targets.cmake)
 
 set( OpenRAVE_PYTHON2 @OPT_PYTHON@ )
 set( OpenRAVE_PYTHON3 @OPT_PYTHON3@ )
 
 if (OpenRAVE_PYTHON2)
-  set( OpenRAVE_PYTHON2_INCLUDE_DIRS "@Python2_INCLUDE_DIRS@;@Python2_NumPy_INCLUDE_DIRS@" )
+  find_package(Python 2.7 REQUIRED COMPONENTS Interpreter Development NumPy)
+
+  set( OpenRAVE_PYTHON2_INCLUDE_DIRS "${Python_INCLUDE_DIRS};${Python_NumPy_INCLUDE_DIRS}" )
   set( OpenRAVE_PYTHON2_DIR "${OpenRAVE_ROOT_DIR}/@OPENRAVEPY2_VER_INSTALL_DIR@")
-  mark_as_advanced(OpenRAVE_PYTHON2_DIR)
   set( OpenRAVE_USE_PYBIND11_PYTHON_BINDINGS @USE_PYBIND11_PYTHON_BINDINGS@)
 
   # Backwards compat: Refer to python 2
-  set( OpenRAVE_PYTHON_INCLUDE_DIRS ${OpenRAVE_PYTHON2_INCLUDE_DIRS} )
+  set( OpenRAVE_PYTHON_INCLUDE_DIRS ${OpenRAVE_PYTHON2_INCLUDE_DIRS} CACHE STRING "This is for backward compatibility with deprecated CMake declarations." FORCE)
   set( OpenRAVE_PYTHON_DIR "${OpenRAVE_PYTHON2_DIR}")
-  mark_as_advanced(OpenRAVE_PYTHON_DIR)
 endif()
 
 if (OpenRAVE_PYTHON3)
-  set( OpenRAVE_PYTHON3_INCLUDE_DIRS "@Python3_INCLUDE_DIRS@;@Python3_NumPy_INCLUDE_DIRS@" )
+  find_package(Python3 3.9 REQUIRED COMPONENTS Interpreter Development NumPy)
+
+  set( OpenRAVE_PYTHON3_INCLUDE_DIRS "${Python3_INCLUDE_DIRS};${Python3_NumPy_INCLUDE_DIRS}" )
   set( OpenRAVE_PYTHON3_DIR "${OpenRAVE_ROOT_DIR}/@OPENRAVEPY3_VER_INSTALL_DIR@")
-  mark_as_advanced(OpenRAVE_PYTHON_DIR)
   set( OpenRAVE_USE_PYBIND11_PYTHON3_BINDINGS @USE_PYBIND11_PYTHON3_BINDINGS@)
+
+  # Set these values if Python2 is not present.
+  if (NOT OpenRAVE_PYTHON2)
+    set( OpenRAVE_PYTHON_INCLUDE_DIRS ${OpenRAVE_PYTHON3_INCLUDE_DIRS} CACHE STRING "This is for backward compatibility with deprecated CMake declarations." FORCE)
+    set( OpenRAVE_PYTHON_DIR "${OpenRAVE_PYTHON3_DIR}")
+  endif()
 endif()
+if (OpenRAVE_USE_PYBIND11_PYTHON_BINDINGS OR OpenRAVE_USE_PYBIND11_PYTHON3_BINDINGS)
+  set(PYBIND11_FINDPYTHON ON)
+  find_package(pybind11 REQUIRED CONFIG)
+else()
+  find_package(Boost ${OpenRAVE_Boost_VERSION} REQUIRED COMPONENTS python)
+endif()
+set_and_check(OpenRAVE_PYTHON_DIR "${OpenRAVE_PYTHON_DIR}")
+
+# Backward compatibility only.
+get_target_property(libopenrave_LOCATION OpenRAVE::libopenrave LOCATION)
+get_target_property(libopenrave-core_LOCATION OpenRAVE::libopenrave-core LOCATION)
+set(OpenRAVE_CORE_LIBRARIES "${libopenrave_LOCATION};${libopenrave-core_LOCATION}")
+
+add_library(libopenrave ALIAS OpenRAVE::libopenrave)
+add_library(libopenrave-core ALIAS OpenRAVE::libopenrave-core)
+
+find_package(Eigen3 3.3.7 REQUIRED)
 
-set( OpenRAVE_Boost_VERSION "@Boost_MAJOR_VERSION@.@Boost_MINOR_VERSION@")
-set( OpenRAVE_BLA_VENDOR "@BLA_VENDOR@" )
```

Comment:
I think OpenRAVE_BLA_VENDOR has to be retained to use the same BLA_VENDOR in controllercommon etc.

(I think I suggested adding `-DBLA_VENDOR=Intel10_64_dyn` to cmakeargs, what happened... (I rememeber something like libmkl_rt.so overhead but it is not concluded yet))

Reply:
I added it back, please reply if you have any concerns

Reply:
No issues. It was not possible for me to know what BLA_VENDOR does since there are no comments and nothing appears to consume the variable... Maybe a bit of comment is good.